tests/objcgentest/ObjCGenErrWarnTests.cs

```diff
@@ -0,0 +1,138 @@
+		{
+			Assert.That (warnsToSearch.Length, Is.EqualTo (shouldFindWarn.Length), $"{nameof (warnsToSearch)} array length must match {nameof (shouldFindWarn)}'s");
+			string valid = Xamarin.Cache.CreateTemporaryDirectory ();
+			var warnerrAssembly = ObjCGeneratorTest.Helpers.Universe.Load ("managedwarn");
+			Assert.NotNull (warnerrAssembly, "warnerrAssembly");
+
+			try {
+				using (var ms = new MemoryStream ()) {
+					var stdout = Console.Out;
+					var stderr = Console.Error;
+					var sw = new StreamWriter (ms);
+
+					Console.SetOut (sw);
+					Console.SetError (sw);
+
+					Driver.Main2 (new string [] { $"-o={valid}", "-c", warnerrAssembly.Location });
+					sw.Flush ();
+					ms.Position = 0;
+					var sr = new StreamReader (ms);
+					var output = sr.ReadToEnd ();
+
+					Console.SetOut (stdout);
+					Console.SetError (stderr);
+
+					var arrCount = warnsToSearch.Length;
+					for (int i = 0; i < arrCount; i++) {
+						if (shouldFindWarn [i])
+							Assert.That (output, Does.Contain (warnsToSearch [i]), $"Did not find: {warnsToSearch [i]}");
+						else
+							Assert.That (output, Does.Not.Contain (warnsToSearch [i]), $"Did find: {warnsToSearch [i]}");
+					}
+				}
+			} finally {
+				Directory.Delete (valid, true);
+			}
+		}
+
+		// We usually are in bin/debug|release
+		static readonly string XcodeFolderPath = $"{AppDomain.CurrentDomain.BaseDirectory}../../xcode";
+		static readonly string MonoMsBuildPath = "/Library/Frameworks/Mono.framework/Versions/Current/Commands/msbuild";
+		static readonly string MonoPath = "/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono";
+		static readonly string XcodebuildPath = "/usr/bin/xcodebuild";
+
+		[Test]
+		[TestCase ("NoInitInSubclassTest", "ConstructorsLib", "NoInitInSubclassTest", "NoInitInSubclassTest", Configuration.Debug, "error: 'initWithId:' is unavailable", true)]
+		public void XcodeBuildErrorTest (string directoryTest, string csprojName, string xcodeprojName, string xcodeprojTarget, Configuration config, string errorToSearch, bool cleanOutputDirs)
+		{
+			Assert.IsTrue (Directory.Exists (XcodeFolderPath), "XcodeFolderPath");
+			Assert.IsTrue (File.Exists (MonoMsBuildPath), "MonoMsBuildPath");
+			Assert.IsTrue (File.Exists (MonoPath), "MonoPath");
+			Assert.IsTrue (File.Exists (XcodebuildPath), "XcodebuildPath");
+
+			try {
+				var testcaseBaseDir = Path.Combine (XcodeFolderPath, directoryTest);
+				var msbuildInfo = new ProcessStartInfo {
+					FileName = MonoMsBuildPath,
+					Arguments = $"/p:Configuration={config} /p:OutputPath=DllOutput {Path.Combine (testcaseBaseDir, csprojName)}.csproj"
+				};
+
+				using (var msbuildProc = Process.Start (msbuildInfo)) {
+					msbuildProc.WaitForExit ();
+					Assert.That (msbuildProc.ExitCode, Is.Zero, "msbuildProc");
+				}
+
+				var eargs = new List<string> {
+					"-c",
+					$"{Path.Combine (testcaseBaseDir, "DllOutput", csprojName)}.dll",
+					$"-o={Path.Combine (testcaseBaseDir, "E4KOutput")}"
```

Comment:
If you put the output in a temporary directory, you can skip the cleanup / directory deletion at the end.

You can tell xcodebuild where to put stuff by passing "CONFIGURATION_BUILD_DIR={tmp path}" as a command-line argument (for an example: https://github.com/mono/Embeddinator-4000/blob/75d1de3/tests/objc-cli/Makefile#L73).